server/app/controllers/applicant/RedirectController.java

```diff
@@ -69,37 +72,71 @@ public CompletionStage<Result> programByName(Http.Request request, String progra
-                return redirect(
-                        controllers.applicant.routes.ApplicantInformationController.edit(
-                            applicant.id))
-                    .withSession(request.session().adding(REDIRECT_TO_SESSION_KEY, request.uri()));
+                return CompletableFuture.completedFuture(
+                    redirect(
+                            controllers.applicant.routes.ApplicantInformationController.edit(
+                                applicantId))
+                        .withSession(
+                            request.session().adding(REDIRECT_TO_SESSION_KEY, request.uri())));
               }
 
-              return redirect(
-                  controllers.applicant.routes.ApplicantProgramReviewController.preview(
-                      applicant.id, programFuture.join().id));
+              return getProgramVersionForApplicant(applicantId, programSlug)
+                  .thenComposeAsync(
+                      (Optional<ProgramDefinition> programForExistingApplication) -> {
+                        // Check to see if the applicant already has an application
+                        // for this program, redirect to program version associated
+                        // with that application if so.
+                        if (programForExistingApplication.isPresent()) {
+                          return CompletableFuture.completedFuture(
+                              redirect(
+                                  controllers.applicant.routes.ApplicantProgramReviewController
+                                      .preview(
+                                          applicantId, programForExistingApplication.get().id())));
+                        }
+
+                        return redirectToActiveProgram(applicantId, programSlug);
+                      },
+                      httpContext.current());
             },
             httpContext.current());
   }
 
+  private CompletionStage<Result> redirectToActiveProgram(long applicantId, String programSlug) {
+    return programService
+        .getActiveProgramDefinitionAsync(programSlug)
+        .thenApplyAsync(
+            (activeProgramDefinition) ->
+                redirect(
+                    controllers.applicant.routes.ApplicantProgramReviewController.preview(
+                        applicantId, activeProgramDefinition.id())),
+            httpContext.current());
+  }
+
+  private CompletionStage<Optional<ProgramDefinition>> getProgramVersionForApplicant(
+      long applicantId, String programSlug) {
+    return applicantService
+        .relevantPrograms(applicantId)
+        .thenApplyAsync(
+            (ImmutableMap<LifecycleStage, ImmutableList<ProgramDefinition>> relevantPrograms) ->
+                relevantPrograms.values().stream()
+                    .flatMap(Collection::stream)
+                    .filter(program -> program.slug().equals(programSlug))
+                    .findAny(),
```

Comment:
In the case where the applicant DOES have existing draft applications in multiple program versions, it seems that this call might return a "random" program version. In reality, we should likely make the version returned here the same as the one from @azizova-leyla  bug fix (e.g. the most recent program version). I also may be misunderstanding things here :).

Reply:
fixed while pairing